test/e2e/framework/scaletest/get-publish-metrics.go

```diff
@@ -0,0 +1,212 @@
+package scaletest
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/microsoft/retina/pkg/telemetry"
+	"github.com/microsoft/retina/test/e2e/common"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	metrics "k8s.io/metrics/pkg/client/clientset/versioned"
+)
+
+type GetAndPublishMetrics struct {
+	KubeConfigFilePath          string
+	AdditionalTelemetryProperty map[string]string
+	Labels                      map[string]string
+	OutputFilePath              string
+	stop                        chan struct{}
+	wg                          sync.WaitGroup
+	telemetryClient             *telemetry.TelemetryClient
+	appInsightsKey              string
+}
+
+func (g *GetAndPublishMetrics) Run() error {
+	if g.appInsightsKey != "" {
+		telemetry.InitAppInsights(g.appInsightsKey, g.AdditionalTelemetryProperty["retinaVersion"])
+
+		telemetryClient, err := telemetry.NewAppInsightsTelemetryClient("retina-scale-test", g.AdditionalTelemetryProperty)
+		if err != nil {
+			return errors.Wrap(err, "error creating telemetry client")
+		}
+
+		g.telemetryClient = telemetryClient
+	}
+
+	g.stop = make(chan struct{})
+	g.wg.Add(1)
+
+	go func() {
```

Comment:
if the caller never calls stop there is a potential for goroutine leak here, maybe have this be a blocking call with a loop and leave concurrency up to caller